src/models/invites-ignorer.ts

```diff
@@ -71,6 +71,17 @@ export enum PolicyScope {
  * our data structures.
  */
 export class IgnoredInvites {
+    // A lock around method `getOrCreateTargetRoom`.
+    // Used to ensure that only one async task of this class
+    // is creating a new target room and modifying the
+    // `target` property of account key `IGNORE_INVITES_POLICIES`.
+    private getOrCreateTargetRoomPromise: Promise<Room> | null = null;
+
+    // A lock around method `withIgnoreInvitesPoliciesLock`.
+    // Used to ensure that only one async task of this class is
+    // modifying `IGNORE_INVITES_POLICIES` at any point in time.
+    private withIgnoreInvitesPoliciesPromise: Promise<{}> = Promise.resolve({});
```

Comment:
these look like functions if the reader isn't extremely careful: can we call them something like `targetRoomLock` instead?